### PR TITLE
[`chore`] Add test for MatryoshkaLoss

### PR DESCRIPTION
**Pull request by fpgmaas**
`losses/MatryoshkaLoss.py` currently has 27% test coverage, this PR raises that to 70%.

### CODE REVIEW

**Comment by tomaarsen** [comment]
The overall test here is a bit suboptimal, because MatryoshkaLoss *also* overrides the forward method, so the mocking prevents that from working normally:
image

Which is why these don't get used. With other words, I think the mocking is actually somewhat "breaking" the loss, as the ForwardDecorator is supposed to result in shrunk embedding dimensions being returned, but that isn't happening due to the mocking. I tried to play around with this a bit to change up what I'm mocking (e.g. trying to mock the underlying Transformer rather than the full SentenceTransformer class), but I couldn't get that to work well.

In short, I think I'm leaning towards not accepting this as-is.

- Tom Aarsen
